internal/tiltfile/print/print.go

```diff
@@ -0,0 +1,56 @@
+package print
+
+import (
+	"errors"
+
+	"go.starlark.net/starlark"
+
+	"github.com/tilt-dev/tilt/internal/tiltfile/starkit"
+	"github.com/tilt-dev/tilt/pkg/logger"
+)
+
+// Additional functions for print output.
+type Extension struct{}
+
+func NewExtension() Extension {
+	return Extension{}
+}
+
+func (Extension) OnStart(env *starkit.Environment) error {
+	err := env.AddBuiltin("warn", warn)
+	if err != nil {
+		return err
+	}
+	err = env.AddBuiltin("fail", fail)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func fail(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var msg string
+	err := starkit.UnpackArgs(thread, fn.Name(), args, kwargs, "msg", &msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return nil, errors.New(msg)
+}
+
+func warn(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var msg string
+	err := starkit.UnpackArgs(thread, fn.Name(), args, kwargs, "msg", &msg)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, err := starkit.ContextFromThread(thread)
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Get(ctx).Warnf("%s", msg)
```

Comment:
I assume the fact of using a `Warnf` here means that this message will show up as a Warning in the alerts tab in the UI, and is not just another log line, right?

Is it worth verifying that in the tests? (Idk if there's an easy way in `print_test.go`, but there's definitely an existing way to verify warnings in `tiltfile_test.go`.)

Reply:
i changed the test to only look at warning logs

here's how tiltfile_test.go warning assertion works under the hood:
https://github.com/tilt-dev/tilt/blob/master/internal/tiltfile/tiltfile_test.go#L5476
it's not that much different than what's here...

yes, right now this will show up in the alerts tab AND be another log line, though we've been talking about alternative UX in the new overview